.github/workflows/platform-test-images.yml

```diff
@@ -7,97 +7,280 @@ on:
       platforms:
         type: string
-        default: '["firecracker", "kvm", "openstack", "tofu"]'
+        default: '["kvm", "openstack", "tofu"]'
         description: 'JSON array of platforms to build'
   # triggered manually
   workflow_dispatch:
     inputs:
       gl_version:
         type: string
         default: nightly
-      force_build:
+      build:
+        type: boolean
+        default: true
+        description: 'Build images'
+      push:
         type: boolean
         default: false
-      force_release:
+        description: 'Push images'
+      push_latest:
         type: boolean
         default: false
+        description: 'Push latest tag'
+      archs:
+        type: string
+        default: '["arm64", "amd64"]'
+        description: 'JSON array of architectures to build'
       platforms:
         type: string
-        default: '["firecracker", "kvm", "openstack", "tofu"]'
+        default: '["kvm", "openstack", "tofu"]'
         description: 'JSON array of platforms to build'
+
+env:
+  IMAGE_REGISTRY: ghcr.io/gardenlinux/gardenlinux
+  IMAGE_BASE: platform-test
+  DEFAULT_ARCHS: '["amd64", "arm64"]'
+  DEFAULT_PLATFORMS: '["kvm", "openstack", "tofu"]'
+
 jobs:
-  images:
+  generate_matrix:
+    name: Generate Matrix
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        shell: bash
+    outputs:
+      build_base_matrix: ${{ steps.set-matrix.outputs.build_base_matrix }}
+      build_matrix: ${{ steps.set-matrix.outputs.build_matrix }}
+      push_matrix: ${{ steps.set-matrix.outputs.push_matrix }}
+      archs: ${{ steps.set-matrix.outputs.archs }}
+    steps:
+      - id: set-matrix
+        run: |
+          # Use inputs if provided, otherwise use defaults
+          ARCHS='${{ inputs.archs || env.DEFAULT_ARCHS }}'
+          PLATFORMS='${{ inputs.platforms || env.DEFAULT_PLATFORMS }}'
+
+          # Create build_base matrix (arch only)
+          BUILD_BASE_MATRIX=$(jq -n \
+            --arg archs "$ARCHS" \
+            '{
+              "arch": ($archs | fromjson)
+            }' | jq -c)
+          
+          # Create build matrix (arch x platform)
+          BUILD_MATRIX=$(jq -n \
+            --arg archs "$ARCHS" \
+            --arg platforms "$PLATFORMS" \
+            '{
+              "arch": ($archs | fromjson),
+              "platform": ($platforms | fromjson)
+            }' | jq -c)
+          
+          # Create push matrix (platforms only)
+          PUSH_MATRIX=$(jq -n \
+            --arg platforms "$PLATFORMS" \
+            '{
+              "platform": ($platforms | fromjson)
+            }' | jq -c)
+          
+          echo "build_base_matrix=${BUILD_BASE_MATRIX}" >> $GITHUB_OUTPUT
+          echo "build_matrix=${BUILD_MATRIX}" >> $GITHUB_OUTPUT
+          echo "push_matrix=${PUSH_MATRIX}" >> $GITHUB_OUTPUT
+          echo "archs=${ARCHS}" >> $GITHUB_OUTPUT
+  build_base:
+    # build if gl_version is not latest and requested (build=true) or triggered by push
```

Comment:
nit-pick: instead of replicating the if condition, this commend should explain why we're doing this. I assume this is about avoiding unnecessary work via caching?

Reply:
https://github.com/gardenlinux/gardenlinux/pull/2835/commits/85fe585e0db40ac18df07d4c4d65f3c0a087c37a fixed the comment